packages/react/src/Tooltip/Tooltip.tsx

```diff
@@ -257,9 +260,12 @@ function Tooltip({direction = 'n', children, className, text, noDelay, align, wr
     wrap && 'tooltipped-multiline',
   )
   return (
-    <TooltipBase role="tooltip" aria-label={text} {...rest} className={classes}>
-      {children}
-    </TooltipBase>
+    // This provider is used to check if an icon button is wrapped with tooltip or not.
+    <TooltipContext.Provider value={{tooltipId}}>
```

Comment:
Small nit: the inline object here will cause context consumers to re-render each time this component renders since the identity will change. It could be worth memoizing the context value (but also maybe not 🤷)

Reply:
Great comment! Updated, memoized both contexts 🔥